packages/ui/src/fontManager.ts

```diff
@@ -0,0 +1,253 @@
+                    return result.data;
+                }
+                // got an unexpected result
+                logger.warn(
+                    `failed to downloaded font ${fontName} due to an unexpected font data`,
+                    {
+                        method: "_downloadAndCacheFont",
+                        data: {
+                            error: result.data ? result.data : "empty font data"
+                        }
+                    }
+                );
+                return null;
+            })
+            .catch(err => {
+                logger.warn(`Failed to download font ${fontName}`, {
+                    method: "_downloadFont",
+                    data: {
+                        error: err
+                    }
+                });
+                return null;
+            });
+    }
+
+    private _injectFontRawStyle(fontName: string, fontData: string): void {
+        try {
+            const style = document.createElement("style");
+            style.innerHTML = fontData;
+            (document.head || document.getElementsByTagName("head")[0]).appendChild(style);
+            logger.info(`font "${fontName}" raw data style was injected`, {
+                method: "_injectFontRawStyle"
+            });
+        } catch (err) {
+            logger.warn(`Failed to inject font ${fontName} data to core player style.`, {
+                method: "_injectFontRawStyle",
+                data: {
+                    error: err
+                }
+            });
+        }
+    }
+
+    private _loadFontFromLocalStorage(fontName: string): string | null {
+        try {
+            return localStorage.getItem(`${FontKeyPrefix}${fontName}`);
+        } catch (err) {
+            logger.warn(
+                `Failed to load font "${fontName}" data, key: ${FontKeyPrefix}${fontName} from localStorage`,
+                {
+                    method: "_loadFontFromLocalStorage",
+                    data: {
+                        error: err
+                    }
+                }
+            );
+            return null;
+        }
+    }
+
+    private _saveFontToLocalStorage(fontName: string, fontValue: string): void {
+        try {
+            localStorage.setItem(`${FontKeyPrefix}${fontName}`, fontValue);
+            logger.info(`font "${fontName}" was cached into localStorage`, {
+                method: "_saveFontToLocalStorage"
+            });
+        } catch (err) {
+            logger.warn(`Failed to cache font "${fontName}" into localStorage`, {
+                method: "_saveFontToLocalStorage",
+                data: {
+                    error: err
+                }
+            });
+        }
+    }
+
+    private _isFontLoaded(fontFamily: string): boolean {
+        if (currentFontFamily === "") return false;
+        if (currentFontFamily === fontFamily) {
+            logger.info(`${fontFamily} was already loaded and set.`, {
```

Comment:
no need this part :) it will pollute the logs as it will happen to any plugin that will be loaded